src/dataflow/components/tools/dataflow-tool.tsx

```diff
@@ -70,52 +72,68 @@ export default class DataflowToolComponent extends BaseComponent<IProps, IState>
-    if (problemWorkspace.primaryDocumentKey) {
-      const primaryDocument = documents.getDocument(problemWorkspace.primaryDocumentKey);
-      if (primaryDocument) {
-        // get snapshot of DocumentContent
-        const contentSnapshot = cloneDeep(getSnapshot(primaryDocument.content));
-        // make a new DocumentContentModel from the snapshot
-        const documentContent = DocumentContentModel.create(contentSnapshot);
-        // find the program tile (should only be 1) and apply the program run info
-        documentContent.tileMap.forEach(tile => {
-          if (tile.content.type === "Dataflow") {
-            const programContent = tile.content as DataflowContentModelType;
-            programContent.setProgramRunId(id);
-            programContent.setProgramStartEndTime(startTime, endTime);
-            programContent.setRunningStatus(endTime);
-          }
-        });
-        const originTitle = primaryDocument.title
-                              ? { originTitle: primaryDocument.title }
-                              : undefined;
-        const properties: IDocumentProperties = { dfRunId: id, ...originTitle };
-        if (datasetName.length > 0) {
-          properties.dfHasData = "true";
-        }
-        // create and load the new document
-        const params: ICreateOtherDocumentParams = {
-                title: datasetName || `${primaryDocument.title}-${getLocalTimeStamp(Date.now())}` ,
-                properties,
-                content: JSON.parse(documentContent.publish())
-              };
-        const newPersonalDocument = await db.createPersonalDocument(params);
-        if (newPersonalDocument) {
-          problemWorkspace.toggleComparisonVisible({ override: false });
-          problemWorkspace.setPrimaryDocument(newPersonalDocument);
+    const document = this.getDocument();
+    if (document) {
+      // get snapshot of DocumentContent
+      const contentSnapshot = cloneDeep(getSnapshot(document.content));
+      // make a new DocumentContentModel from the snapshot
+      const documentContent = DocumentContentModel.create(contentSnapshot);
+      // find the program tile (should only be 1) and apply the program run info
+      documentContent.tileMap.forEach(tile => {
+        if (tile.content.type === "Dataflow") {
+          const programContent = tile.content as DataflowContentModelType;
+          programContent.setProgramRunId(id);
+          programContent.setProgramStartEndTime(startTime, endTime);
+          programContent.setRunningStatus(endTime);
         }
+      });
+      const dfProgramId = document.key;
+      const originTitle = document.title
+                            ? { originTitle: document.title }
+                            : undefined;
+      const properties: IDocumentProperties = { dfProgramId, dfRunId: id, ...originTitle };
+      if (datasetName.length > 0) {
+        properties.dfHasData = "true";
       }
```

Comment:
Random comment, but these seems like two different ways of attaching optional properties to the `properties` object. Any reason not to replace lines 106-108 with  `if (document.title) { properties.originTitle = document.title }` for consistency?

Reply:
Good point -- fixed.